### PR TITLE
Set properties feature

### PR DESCRIPTION
**Pull request by joerg**
Updated the properties reader to also set properties if some are passed. Pretty straight forward.
Let me know what you think about it...

### CODE REVIEW

File: lib/artifactory/resources/artifact.rb [hunk 1]
```diff
@@ -620,6 +631,45 @@ def upload_from_archive(repo, remote_path, properties = {})
...
+    # @example Set properties for an artifact
+    #   artifact.set_properties({ prop1: 'value1', 'prop2' => 'value2' })
+    #
+    # @param [Hash<String, Object>] props
+    #   A hash of properties and corresponding values to set for the artifact
+    #
+    # @return [Hash]
+    #   the parsed JSON response from the server
+    #
+    def set_properties(props)
```
Comment (schisamo) [inline lib/artifactory/resources/artifact.rb]: Let's use the full name of `properties` for this parameter.

File: lib/artifactory/resources/artifact.rb [hunk 2]
```diff
@@ -620,6 +631,45 @@ def upload_from_archive(repo, remote_path, properties = {})
...
+      safe_properties = props.map { |p, v| "#{url_safe(p)}=#{url_safe(v)}" }.join(";")
+      artifact_url_with_props = "#{File.join("/api/storage", relative_path)}?properties=#{safe_properties}"
```
Comment (schisamo) [inline lib/artifactory/resources/artifact.rb]: We can simplify this by taking advantage of the `Artifactory::Resource::Base#to_matrix_properties ` helper:

```ruby
matrix   = to_matrix_properties(properties)
endpoint = File.join("/api/storage", relative_path) + "?properties=#{matrix}"

client.put(endpoint, nil)
```

**Comment by joerg** [comment]
Yeah, sorry. Actually I used to_matrix_properties in my small playground script to test Artifactory but somehow this did not make it into my PR. Well, I updated everything and also tested it with our Artifactory instance.

**Comment by schisamo** [comment]
@joerg Thanks again for the contribution...this as released as part of 2.8.0.

**Pull request merged**